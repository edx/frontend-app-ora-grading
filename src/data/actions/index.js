import { StrictDict } from 'utils';

import app from './app';
import grading from './grading';
import submissions from './submissions';

export default StrictDict({
  app,
  grading,
  submissions,
});
