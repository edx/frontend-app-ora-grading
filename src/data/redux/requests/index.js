export { actions, reducer } from './reducer';
