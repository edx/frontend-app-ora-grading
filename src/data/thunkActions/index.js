import { StrictDict } from 'utils';

import app from './app';
import grading from './grading';

export default StrictDict({
  app,
  grading,
});
