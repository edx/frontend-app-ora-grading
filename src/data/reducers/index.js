import { combineReducers } from 'redux';

import app from './app';
import submissions from './submissions';

/* istanbul ignore next */
const rootReducer = combineReducers({
  app,
  submissions,
});

export default rootReducer;
