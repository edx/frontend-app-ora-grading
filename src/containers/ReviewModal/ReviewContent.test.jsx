import React from 'react';
import { shallow } from 'enzyme';

import { selectors } from 'data/redux';
import {
  ReviewContent,
  mapStateToProps,
} from './ReviewContent';

jest.mock('data/redux', () => ({
  selectors: {
    app: {
      showRubric: (...args) => ({ showRubric: args }),
    },
  },
}));
jest.mock('containers/ResponseDisplay', () => 'ResponseDisplay');
jest.mock('containers/Rubric', () => 'Rubric');

jest.useFakeTimers('modern');

describe('ReviewContent component', () => {
  describe('component', () => {
    describe('render tests', () => {
      test('snapshot (show rubric)', () => {
        expect(shallow(<ReviewContent />)).toMatchSnapshot();
      });
      test('snapshot (hide rubric)', () => {
        expect(shallow(<ReviewContent showRubric />)).toMatchSnapshot();
      });
    });
  });
  describe('mapStateToProps', () => {
    let mapped;
    const testState = { some: 'test-state' };
    beforeEach(() => {
      mapped = mapStateToProps(testState);
    });
    test('showRubric loads from app.showRubric', () => {
      expect(mapped.showRubric).toEqual(selectors.app.showRubric(testState));
    });
  });
});
