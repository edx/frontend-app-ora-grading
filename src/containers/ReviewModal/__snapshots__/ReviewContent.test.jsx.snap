// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewContent component component render tests snapshot (hide rubric) 1`] = `
<div
  className="content-block"
>
  <Row
    className="flex-nowrap"
  >
    <Col>
      <ResponseDisplay />
    </Col>
    <Rubric />
  </Row>
</div>
`;

exports[`ReviewContent component component render tests snapshot (show rubric) 1`] = `
<div
  className="content-block"
>
  <Row
    className="flex-nowrap"
  >
    <Col>
      <ResponseDisplay />
    </Col>
  </Row>
</div>
`;
